OpenRA.Mods.Common/Traits/World/EditorActorPreview.cs

```diff
@@ -80,16 +80,19 @@ public EditorActorPreview(WorldRenderer worldRenderer, string id, ActorReference
 
 			// Bounds are fixed from the initial render.
 			// If this is a problem, then we may need to fetch the area from somewhere else
-			var r = previews
+			var rs = previews
 				.SelectMany(p => p.Render(worldRenderer, CenterPosition))
 				.Select(rr => rr.PrepareRender(worldRenderer));
 
-			if (r.Any())
+			Game.Renderer.WorldModelRenderer.BeginFrame();
```

Comment:
This really doesn't seem right.  The ModelRenderer shouldn't be needed for the bounds calculation!

Reply:
Calling BeginFrame/EndFrame from anywhere but the world renderer also feels quite wrong.  We should instead try and defer `PrepareRender` until `Render` (where we should be inside a frame), and then identify what plumbing needs to change so that we can immediately calculate Bounds on the non-finalized renderables.

Reply:
The ModelRenderer doesn't know you're only borrowing it to calculate bounds and not actually render anything. I think it should be well within it's rights to assume you're calling RenderAsync with the intent of actually rendering stuff.

An alternative would be to silently drop any rendering requests if you're not in a frame (i.e. don't queue a render action or create the sheets required) but still give you the rendering info but that seems like a horrible gotcha.

Reply:
The standard `IRenderable` actually isn't actually expected to know how to render itself.  It needs to call `PrepareRender` to get an `IFinalizedRenderable` first for that.

If we defer the `PrepareRender` until we actually need it we can then use 493c9aab79d03f7f83abd8ebd2b4e5ffec28de1f (rescued out of my reflog) to query the bounds directly from the non-final renderable... hopefully.

Reply:
(note that I edited the comment above - we don't need to cache anything, just remove the PrepareRender)

Reply:
In that case, switching to "be prepared to not render when not in a frame" sounds fine - I'll switch to that.